### PR TITLE
fix potential out-of-bound access when receiving LOGGING_DATA* msg

### PR DESCRIPTION
**Pull request by BOB4Drone**
FIX [issue #10037](https://github.com/mavlink/qgroundcontrol/issues/10037)

### CODE REVIEW

File: src/Vehicle/Vehicle.cc [hunk 1]
```diff
@@ -3421,17 +3421,26 @@ void Vehicle::_handleMavlinkLoggingData(mavlink_message_t& message)
 {
     mavlink_logging_data_t log;
     mavlink_msg_logging_data_decode(&message, &log);
-    emit mavlinkLogData(this, log.target_system, log.target_component, log.sequence,
+    if (static_cast<size_t>(log.length) > sizeof(log.data)) {
+            qWarning() << "Invalid length for LOGGING_DATA, discarding." << log.length;
+        } else {
```
Comment (bkueng) [inline src/Vehicle/Vehicle.cc:3426]: Can you fix the indentation (4 spaces)?

Comment (BOB4Drone) [reply src/Vehicle/Vehicle.cc]: fixed it. check plz

File: src/Vehicle/Vehicle.cc [hunk 2]
```diff
@@ -3421,17 +3421,25 @@ void Vehicle::_handleMavlinkLoggingData(mavlink_message_t& message)
 {
     mavlink_logging_data_t log;
     mavlink_msg_logging_data_decode(&message, &log);
-    emit mavlinkLogData(this, log.target_system, log.target_component, log.sequence,
-                        log.first_message_offset, QByteArray((const char*)log.data, log.length), false);
+    if (static_cast<size_t>(log.length) > sizeof(log.data)) {
+            qWarning() << "Invalid length for LOGGING_DATA, discarding." << log.length;
```
Comment (bkueng) [inline src/Vehicle/Vehicle.cc:3425]: Sorry, I don't like to be nitpicky about this, but this is still off (below as well).

**Comment by bkueng** [comment]
I fixed the remaining whitespace issue and applied it: https://github.com/mavlink/qgroundcontrol/commit/f6a410f38c9a340f0b64e6f3c48a3566dfd8e8da.
Thanks for contributing.

**Comment by BOB4Drone** [comment]
Oh, there was a misunderstanding. I'm glad it's fixed well.

**Pull request closed without merging**